rosidl_generator_py/rosidl_generator_py/generate_py_impl.py

```diff
@@ -0,0 +1,207 @@
+# Copyright 2014-2016 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections import defaultdict
+import os
+
+from rosidl_cmake import convert_camel_case_to_lower_case_underscore
+from rosidl_cmake import expand_template
+from rosidl_cmake import get_newest_modification_time
+from rosidl_cmake import read_generator_arguments
+from rosidl_parser import parse_message_file
+from rosidl_parser import parse_service_file
+
+
+def generate_py(generator_arguments_file, typesupport_impl, typesupport_impls):
+    args = read_generator_arguments(generator_arguments_file)
+    typesupport_impls = typesupport_impls.split(';')
+
+    template_dir = args['template_dir']
+    type_support_impl_by_filename = {
+        '_%s_s.ep.{0}.c'.format(impl): impl for impl in typesupport_impls
+    }
+    mapping_msgs = {
+        os.path.join(template_dir, '_msg.py.template'): ['_%s.py'],
+        os.path.join(template_dir, '_msg_support.c.template'): ['_%s_s.c'],
+        os.path.join(template_dir, '_msg_support.entry_point.c.template'):
+        type_support_impl_by_filename.keys(),
+    }
+
+    mapping_srvs = {
+        os.path.join(template_dir, '_srv.py.template'): ['_%s.py'],
+    }
+
+    for template_file in mapping_msgs.keys():
+        assert os.path.exists(template_file), \
+            'Messages template file "%s" not found' % template_file
```

Comment:
Other code uses single quotes for the user visible part and then double quotes around the string.